homeassistant/components/teslemetry/cover.py

```diff
@@ -201,14 +201,22 @@ async def async_added_to_hass(self) -> None:
     def _handle_stream_update(self, data) -> None:
         """Update the entity attributes."""
 
-        if value := data.get(Signal.FD_WINDOW):
-            self.fd = WindowState.get(value) == "closed"
-        if value := data.get(Signal.FP_WINDOW):
-            self.fp = WindowState.get(value) == "closed"
-        if value := data.get(Signal.RD_WINDOW):
-            self.rd = WindowState.get(value) == "closed"
-        if value := data.get(Signal.RP_WINDOW):
-            self.rp = WindowState.get(value) == "closed"
+        change = False
+        if value := data["data"].get(Signal.FD_WINDOW):
+            self.fd = WindowState.get(value) == "Closed"
+            change = True
+        if value := data["data"].get(Signal.FP_WINDOW):
+            self.fp = WindowState.get(value) == "Closed"
+            change = True
+        if value := data["data"].get(Signal.RD_WINDOW):
+            self.rd = WindowState.get(value) == "Closed"
+            change = True
+        if value := data["data"].get(Signal.RP_WINDOW):
+            self.rp = WindowState.get(value) == "Closed"
+            change = True
+
+        if not change:
```

Comment:
It would be nice if you could also add a test for this

Reply:
This can also be done in a follow up pull request. I'll go ahead with the merge 👍 

../Frenck